subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskExecutionErrorHandlingIntegrationTest.groovy

```diff
@@ -107,20 +114,45 @@ class CachedTaskExecutionErrorHandlingIntegrationTest extends AbstractIntegratio
         """
     }
 
-    def "cache switches off after third error for the current build"() {
+    def "cache switches off after third recoverable error for the current build"() {
         // We require a distribution here so that we can capture
         // the output produced after the build has finished
         executer.requireGradleDistribution()
         executer.withBuildCacheEnabled()
         executer.withStackTraceChecksDisabled()
 
         when:
-        succeeds "assemble", "-Dfail"
+        succeeds "assemble", "-Dfail", "-Drecoverable"
         then:
         output.count("Could not load entry") == 2
         output.count("Could not store entry") == 1
-        output.count("The remote build cache is now disabled because 3 errors were encountered") == 1
-        output.count("The remote build cache was disabled during the build after encountering 3 errors.") == 1
+        output.count("The remote build cache is now disabled because 3 recoverable errors were encountered.") == 1
+        output.count("The remote build cache was disabled during the build because 3 recoverable errors were encountered.") == 1
+
+        expect:
+        succeeds "clean"
+
+        when:
+        succeeds "assemble"
```

Comment:
Could we put this together (`succeeds "clean", "assemble"`)? and add a comment like, `// build cache should be still enabled during next build`